rest/servers/mock/src/main/java/com/connexta/discovery/rest/springboot/mock/controller/DiscoveryController.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.discovery.rest.springboot.mock.controller;
+
+import com.connexta.discovery.rest.models.ContactInfo;
+import com.connexta.discovery.rest.models.ResponseMessage;
+import com.connexta.discovery.rest.models.SystemInfo;
+import com.connexta.discovery.rest.spring.DiscoveryApi;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Iterator;
+import java.util.Optional;
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+
+/** Dummy controller implementation to provide a simple test stub for the discovery API. */
```

Comment:
❓Is this test stub used to test the generated clients, or is it meant to be used by the external clients of this API? In the latter case, should we be hand-coding and source controlling these, or should we let clients use something like rest assured? Or leave this out for now until we have a code generator that can create a stub from the API spec?

Reply:
It ain't meant to be generic right now. It was my first quick and dirty implementation which help test the protocol and the client until such time where we have an actual server implementing the protocol to test against. Not sure this module will remain here over time.